src/types/index.ts

```diff
@@ -376,6 +376,29 @@ export interface TextFileOnChangeEvent {
   latestContent: string;
 }
 
+export interface WatchDirListeners {
+  onChange: WatchDirOnChangeListener;
+  onMoveOrDelete?: WatchDirOnMoveOrDeleteListener;
+  onError: WatchDirOnErrorListener;
+}
+
+export enum FsErrors {
+  NotFound = "NOT_FOUND",
+  AlreadyExists = "ALREADY_EXIST",
+  NotDirectory = "NOT_DIRECTORY",
+  IsDirectory = "IS_DIRECTORY",
+  InvalidPath = "INVALID_PATH",
+  DiskQuotaExceeded = "DISK_QUOTA_EXCEEDED",
+}
+
+export type WatchDirOnErrorListener = (
+  err: Error & { code?: FsErrors.NotFound | FsErrors.NotDirectory },
```

Comment:
Are you sure those errors are being pass through correctly? IIRC they didn't work as expected for file watchers so I left them out. If they do work, let's add the codes to the other watchers as well.

Reply:
I'm not sure how to effectively test that it works so I'll just set the type to `Error` for now.